src/control/cmd/daos_server/storage.go

```diff
@@ -222,7 +217,12 @@ func (cmd *storagePrepareCmd) Execute(args []string) error {
 
 	scanErrors := make(errs, 0, 2)
 
-	if err := cmd.prepNvme(cmd.scs.NvmePrepare, iommuDetected()); err != nil {
+	iommuEnabled, err := sysfs.NewProvider(cmd).IsIOMMUEnabled()
+	if err != nil {
+		return err
+	}
```

Comment:
See, this is kind of silly, right? You know that `NewProvider()` isn't going to return a nil pointer, so there's no reason to check for an error...

Reply:
fixing

Reply:
now that we are returning error from ReadDir as per Kris's change it seems that we do need to check error